Ghidra/Processors/STM/data/languages/ST7.sinc

```diff
@@ -0,0 +1,758 @@
+  nzFlags(result);
+}
+
+:BRES imm8, "#"bit_position is $(NO_PRE) & bit_upper=0x01 & bit_lower=0x01 & bit_position ; imm8 {
+  local address:2 = imm8;
+  local value = *:1 address;
+  value = value & ~(1 << bit_position);
+  *:1 address = value;
+}
+
+:BRES [imm8], "#"bit_position is $(PRE_PIX) & bit_upper=0x01 & bit_lower=0x01 & bit_position ; imm8 {
+  local address:2 = imm8;
+  local target:2 = *:1 address;
+  local value:1 = *:1 target;
+  value = value & ~(1 << bit_position);
+  *:1 target = value;
+}
+
+:BSET imm8, "#"bit_position is $(NO_PRE) & bit_upper=0x01 & bit_lower=0x00 & bit_position ; imm8 {
+  local address:2 = imm8;
+  local value = *:1 address;
+  value = value | (1 << bit_position);
+  *:1 address = value;
+}
+
+:BSET [imm8], "#"bit_position is $(PRE_PIX) & bit_upper=0x01 & bit_lower=0x00 & bit_position ; imm8 {
+  local address:2 = imm8;
+  local target:2 = *:1 address;
+  local value:1 = *:1 target;
+  value = value | (1 << bit_position);
+  *:1 target = value;
+}
+
+:BTJF imm8, "#"bit_position, REL is $(NO_PRE) & bit_upper=0x00 & bit_lower=0x01 & bit_position ; imm8 ; REL {
+  local address:2 = imm8;
+  local value = *:1 address;
+  if (value & (1 << bit_position)) == (1 << bit_position)  goto inst_next;
+  PC = REL;
+}
+
+:BTJF [imm8], "#"bit_position, REL is $(PRE_PIX) & bit_upper=0x00 & bit_lower=0x01 & bit_position ; imm8 ; REL {
+  local address:2 = imm8;
+  local target:2 = *:1 address;
+  local value:1 = *:1 target;
+  if (value & (1 << bit_position)) == (1 << bit_position)  goto inst_next;
+  PC = REL;
+}
+
+:BTJT imm8, "#"bit_position, REL is $(NO_PRE) & bit_upper=0x00 & bit_lower=0x00 & bit_position ; imm8 ; REL {
+  local address:2 = imm8;
+  local value = *:1 address;
+  if (value & (1 << bit_position)) == 0 goto inst_next;
+  PC = REL;
+}
+
+:BTJT [imm8], "#"bit_position, REL is $(PRE_PIX) & bit_upper=0x00 & bit_lower=0x00 & bit_position ; imm8 ; REL {
+  local address:2 = imm8;
+  local target:2 = *:1 address;
+  local value:1 = *:1 target;
+  if (value & (1 << bit_position)) == 0 goto inst_next;
+  PC = REL;
+}
+
+:CALL ADDROP is kind=0x0D ... & ADDROP {
+  PC = inst_next;
+  *:1 SP = PCL;
+  SP = SP - 1;
+  *:1 SP = PCH;
+  SP = SP - 1;
+  PC = ADDROP;
+}
+
+:CALLR REL is $(NO_PRE) & opcode=0xAD ; REL {
+  PC = inst_next;
+  *:1 SP = PCL;
+  SP = SP - 1;
+  *:1 SP = PCH;
+  SP = SP - 1;
+  PC = REL;
+}
+
+:CALLR [imm8] is $(PRE_PIX) & opcode=0xAD ; imm8 {
+  local address:2 = imm8;
+  local target:2 = *:1 address;
+  local offset = sext(*:1 target);
+  PC = inst_next;
+  *:1 SP = PCL;
+  SP = SP - 1;
+  *:1 SP = PCH;
+  SP = SP - 1;
+  PC = PC + offset;
+}
+
+:JR^JCC REL is $(NO_PRE) & JCC ; REL {
+  if !CC goto inst_next;
+  PC = PC + REL;
+}
+
+:JR^JCC [imm8] is $(PRE_PIX) & JCC ; imm8 {
+  local address:2 = imm8;
+  local target:2 = *:1 address;
+  local offset = sext(*:1 target);
+  if !CC goto inst_next;
+  PC = PC + offset;
+}
+
+:CLR BITOP is kind=0x0F ... & BITOP {
+  BITOP = 0;
```

Comment:
setting looks questionable, when it returns the regs, maybe it works, but for the other cases not so sure.  would you want `*:1 BITOP = 0;` for the memory byte location destinations?

goes for the rest of the file.

Reply:
From what I can see in the decompilation window it should work as it is:
<img width="1716" alt="Screenshot 2021-11-14 at 23 14 09" src="https://user-images.githubusercontent.com/235740/141700679-09ea7773-a46c-4fb6-b2c4-9274627570fb.png">